src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy

```diff
@@ -1077,14 +1129,38 @@ class FlowCrudSpec extends HealthCheckSpecification {
                             flowToConflict.destination.vlanId = 0
                             dominantFlow.destination.vlanId = 0
                         },
-                        getError            : { FlowPayload dominantFlow, FlowPayload flowToConflict,
-                                                String operation = "create" ->
-                            errorMessage(operation, dominantFlow, "destination", flowToConflict, "destination")
+                        getErrorMessage            : { FlowPayload dominantFlow, FlowPayload flowToConflict,
+                                                       String operation = "create" ->
+                            "Could not $operation flow"
+                        },
+                        getErrorDescription        : { FlowPayload dominantFlow, FlowPayload flowToConflict,
+                                                       String operation = "create" ->
+                            errorDescription(operation, dominantFlow, "destination", flowToConflict, "destination")
                         }
                 ]
         ]
     }
 
+    @Shared
+    def errorDescription = { String operation, FlowPayload flow, String endpoint, FlowPayload conflictingFlow,
+                             String conflictingEndpoint ->
+        def message = "Requested flow '$conflictingFlow.id' " +
+                "conflicts with existing flow '$flow.id'. " +
+                "Details: requested flow '$conflictingFlow.id' $conflictingEndpoint: " +
+                "switchId=\"${conflictingFlow."$conflictingEndpoint".datapath}\" " +
+                "port=${conflictingFlow."$conflictingEndpoint".portNumber}"
+        if (0 < conflictingFlow."$conflictingEndpoint".vlanId) {
+            message += " vlanId=${conflictingFlow."$conflictingEndpoint".vlanId}";
+        }
+        message += ", existing flow '$flow.id' $endpoint: " +
+                "switchId=\"${flow."$endpoint".datapath}\" " +
+                "port=${flow."$endpoint".portNumber}"
+        if (0 < flow."$endpoint".vlanId) {
+            message += " vlanId=${flow."$endpoint".vlanId}"
+        }
+        return message
+    }
+
     private static Message buildMessage(final CommandData data) {
```

Comment:
there will be a lot of conflicts with https://github.com/telstra/open-kilda/pull/3432 =(